src/ValueParsers/IsoTimestampParser.php

```diff
@@ -163,12 +163,18 @@ private function getPrecision( array $timeParts ) {
 
 		$option = $this->getOption( self::OPT_PRECISION );
 
-		// It's impossible to increase the detected precision via option, e.g. from year to month if
-		// no month is given. If a day is given it can be increased, relevant for midnight.
-		if ( is_int( $option )
-			&& ( $option <= $precision || $precision >= TimeValue::PRECISION_DAY )
-		) {
-			return $option;
+		if ( $option !== null ) {
+			if ( !is_int( $option ) && !ctype_digit( $option ) ) {
+				throw new ParseException( 'Precision must be an integer' );
```

Comment:
"Numerical string" wouldn't be correct. Only integers are accepted, no matter if they are wrapped in a string or not.